src/test/test_addr.c

```diff
@@ -1671,8 +1671,53 @@ test_addr_octal(void *arg)
   ;
 }
 
+#define TEST_ADDR_VALIDITY(a, lis, rv) STMT_BEGIN \
+    tt_int_op(tor_addr_is_valid(a, lis), OP_EQ, rv); \
+  STMT_END;
+
+static void
+test_addr_is_valid(void *arg)
+{
+  (void)arg;
+  tor_addr_t *test_addr = tor_malloc(sizeof(tor_addr_t));
```

Comment:
You're missing tests for Unix addresses, and the unspecified null address.

You can use tor_addr_make_null(AF_UNSPEC) to make an unspecified null address.